src/components/gallery/Gallery.tsx

```diff
@@ -0,0 +1,74 @@
+import React, {useCallback, useState} from 'react';
+import {
+  Image,
+  ImageSourcePropType,
+  StyleSheet,
+  TouchableOpacity,
+  View,
+} from 'react-native';
+import Icon from 'react-native-vector-icons/MaterialIcons';
+import {Indicator} from 'components/gallery/Indicator';
+
+interface GalleryProps {
+  images: ImageSourcePropType[];
+}
+
+export const Gallery: React.FC<GalleryProps> = ({images = []}) => {
+  const [activeImageIdx, setActiveImageIdx] = useState(0);
+
+  const selectImage = useCallback((index: number) => {
+    setActiveImageIdx(index);
+  }, []);
+
+  const showPrevImage = useCallback(() => {
+    selectImage(activeImageIdx === 0 ? images.length - 1 : activeImageIdx - 1);
+  }, [activeImageIdx, images.length, selectImage]);
+
+  const showNextImage = useCallback(() => {
+    selectImage(activeImageIdx === images.length - 1 ? 0 : activeImageIdx + 1);
+  }, [activeImageIdx, images.length, selectImage]);
+
+  return images.length > 0 ? (
```

Comment:
it can be just `images.length > 0 && ...`

Reply:
using `&&` here will result in `false` value returned sometimes instead of a component, this raises type check error